main/src/library/MutDeque.flix

```diff
@@ -31,6 +31,30 @@ pub enum MutDeque[a] {
     case MutDeque(Ref[Array[a]], Ref[Int32], Ref[Int32])
 }
 
+instance Iterable[MutDeque] {
+    pub def iterator(t: MutDeque[a]): Iterator[a] & Impure =
+        let MutDeque(a, f, b) = t;
+        let len = Array.length(deref a);
+        let copy = deref a;
+        let i = f;
+
+        let done = () -> (deref i) == deref b;
+        let next = () -> {
+            if (deref i != deref b) {
+                let j = {
+                    if (deref i == len) 0
+                    else deref i
+                };
+                let x = copy[j];
+                i := j + 1;
+                x
+            } else {
+                bug!("Empty Iterator!")
+            }
+        };
+        Iterator(done, next)
```

Comment:
Should I move this implentation into the `namespace MutDeque` as a `def toIterator`, and then call it here in the `instance Iterable[MutDeque]`? Or is it better to have it here?

Reply:
Yes, lets do that for now.